source/sonata_tech.rst

```diff
@@ -72,7 +72,7 @@ Group column represents the HDF group where the dataset is located under /<popul
     /0                 ``mtype``                       utf8       Mandatory     Defines the morphological type of the node.
     /0                 ``me_combo``                    utf8       Optional      Deprecated Replaced by dynamics_params.
                                                                                 Foreign key to me combo tsv file.
-    /0                 ``synapse_class``               utf8       Mandatory     Defines the synapse type of the node; whether the neuron is inhibitory or excitatory. "EXC" or "INH".
+    /0                 ``synapse_class``               utf8       Mandatory     Defines the synapse type of the node; whether the neuron is inhibitory or excitatory. "EXC" or "INH".(or "MOD" - Define a third class of synapse (other than excitatory and inhibitory class) which is neuromodulatory.)
```

Comment:
```suggestion
    /0                 ``synapse_class``               utf8       Mandatory     Defines the synapse type of the node; whether the neuron is inhibitory, excitatory, or neuromodulatory. "EXC", "INH", or "MOD".
```
For better wording…

Reading up a bit on this (as I'm not a neuroscientist), it seems to me that modulators enhance synaptic response, but are not necessarily a synapse type?

Reply:
I think "MOD" is too general. Would suggest "Cholinergic." It will prevent confusion down the line once additional neuromodulatory types are added.

Reply:
@vignan21 can you update accordingly ?
@MWolfR @romani79  no other comment ?

Reply:
I think the changes made by @matz-e and @dkeller9 seem reasonable.

Reply:
@vignan21 I like your suggest. The synapse class remains MOD, but then we can have different synapse class for each neuromodulator. We can start with M1 for Acetylcholine. I think this is enough for now. We can always append new type/neuromodulator depending on our needs. Listing "all" the neuromodulators and their combinations may be too much. Let's start simple.

Reply:
@romani79 not sure I follow what you are proposing. If the synapse class is MOD then where would the different synapse class for each modulator be specified?

Reply:
I believe that the general consensus is to distinguish the synapse class based on the neurotransmitter released. So, we have EXC (=GLU), INH (=GABA), ACH, DOP... Do we want to keep the three-letters abbreviations or aim for something more flexible?

Reply:
@romani79 > aim for something more flexible?

Could you describe what you have in mind by `flexible`?

Reply:
My two cents:
This classification proposal is a bit odd to me. glutamate is not the only excitatory neurotransmitter, and gaba not the only inhibitory one. But it is implicitly assumed as such whereas all the rest synapse classes are specific to the neurotransmitter released. 
The initial proposal of having three groups seemed more reasonable to me. An additional property could help specify the neurotransmitter released, if that would work for you.

Reply:
> Could you describe what you have in mind by `flexible`?

@mgeplf Just allowing any string rather than 3-letters acronyms.

Reply:
@eleftherioszisis as I mentioned in the call, to better capture the biology, we should have a least two parameters: neurotransmitter and the postsynaptic effect (E, I).
Is it something you can implement with backwards compatibility?

Reply:
This is not something for me to decide. That was just my opinion on not mixing a generic group (e.g. EXC, INH) with neurotransmitter-specific ones (e.g. ACH) in the same property.
You'll need to make a clear use case and coordinate with HPC because ultimately it is the simulator that will handle these.

Reply:
My proposal is to go for the parameter set that is closest to biology, that is neurotransmitter (glutamate, gaba, acetylcholine...) and the postsynaptic effect (E, I).
An even more accurate strategy is to specify postsynaptic receptors. It may sound too much, but there are several benefits. First, we avoid the use of Exc and Inh that may generate ambiguity with the current implementation. In addition, Exc or Inh effect also depends on the postsynaptic membrane voltage. What really matters is the channels in the postsynaptic terminals. This would be a pretty general implementation. Then a postsynaptic receptor is associated with a mod file (e.g. AMPA/NMDA, GABAA/GABAB).

Reply:
> > Could you describe what you have in mind by `flexible`?
> 
> @mgeplf Just allowing any string rather than 3-letters acronyms.

Sure, that's no problem.

Reply:
@romani79 
> An even more accurate strategy is to specify postsynaptic receptors

To be clear, what you're proposing is that, for each connection, there are two attributes with properties:
```
neurotransmitter: One of glutamate, gaba, acetylcholine, (potentially more to be defined)
postsynaptic_receptors: One of AMPA/NMDA, GABAA/GABAB, (potentially more to be defined)
```

I can see how the postsynaptic_receptors modifies the simulation (ie: choice of mod file), what is the impact of the first, concretely?

That seems reasonable, but it would be worth having people from the simulation side say it's viable: @jamesgking?

Reply:
> what is the impact of the first, concretely?

In principle it is redundant if we assume that for a given synapses the presynaptic terminal releases the neurotransmitter compatible with the postsynaptic receptors. This is a fair assumption and we avoid useless complications.
I would ask the other scientists if I am missing obvious exceptions for which we should specify presynaptic neurotransmitters.

Reply:
Apparently, I created a bit of mess here. The topic is about node file, while I was discussion more in terms of edge file. During the meeting, we commented that the column on Exc/Inh in the **node** file may be redundant if we have the postsynapse receptor information in the **edge** file.

Reply:
@wvangeit the redundancy is at the level of neurotransmitter release, but both receptor and model files are essential information as you pointed out. The proposal we had in the meeting is to store the information on receptor in the edge file, and the information on which mod file to use outside edge file (e.g. recipe, Neurodamus).

Reply:
> The proposal we had in the meeting is to store the information on receptor in the edge file, and the information on which mod file to use outside the edge file (e.g. recipe, Neurodamus).

I don't think I like this proposal. The format is a "circuit model definition", so which mod file to load should be defined in the circuit as we do with morphologies, or with the e-model to load. That is the information that is needed to the simulator.